backend/substrapp/tasks/tasks.py

```diff
@@ -80,22 +79,26 @@ def _get_model(model):
     return model_content
 
 
-def get_model(subtuple):
-    model = subtuple.get('model')
-    if model:
-        return _get_model(model)
+def get_model(testtuple):
+    traintuple_hash = testtuple.get('traintupleKey')
+    if traintuple_hash:
+        return _get_model(traintuple_hash)
     else:
         return None
 
 
-def get_models(subtuple):
-    input_models = subtuple.get('inModels')
+def get_models(traintuple):
+    input_models = traintuple.get('inModels')
     if input_models:
-        return [_get_model(item) for item in input_models]
+        return [_get_model(item['traintupleKey']) for item in input_models]
     else:
         return []
 
 
+def get_traintuple_metadata(traintupleKey):
```

Comment:
I think we should have `traintuple_hash` here :)

Reply:
Fixed https://github.com/SubstraFoundation/substra-backend/pull/33/commits/d11f3b4803767841bdc46f22aa87492c46845cfc